### PR TITLE
Add NameAndTagKeys as key in 2-level dictionary.

### PR DESCRIPTION
**Pull request by victlu**
Frame the key to our 2-level dictionary to include Name (of View/Instrument) + TagKeys.

For now, the Name is set to the Instrument.Name.

Need this ability for future View work where the Name may be a View name.

### CODE REVIEW

File: src/OpenTelemetry/Metrics/AggregatorStore.cs [hunk 1]
```diff
@@ -105,27 +107,34 @@ internal IAggregator[] FindMetricAggregators(ReadOnlySpan<KeyValuePair<string, o
                 Array.Sort<string, object>(tagKey, tagValue);
             }
 
+            string name = this.instrument.Name;
```
Comment (cijothomas) [inline src/OpenTelemetry/Metrics/AggregatorStore.cs:110]: the instrument name (and in future view name as well), will be part of the Aggregator anyway. and we have the ability to map an incoming measurement to multiple aggregators. So why do we have to add the viewname/instrumentname to the key?

Comment (victlu) [reply src/OpenTelemetry/Metrics/AggregatorStore.cs:110]: An instrument may contribute to multiple views.

Comment (victlu) [reply src/OpenTelemetry/Metrics/AggregatorStore.cs:110]: Also, I do not agree that an Instrument is part of the Aggregator. An Aggregator is just an aggregator without regards to which instrument contributes values to it.

However, an Aggregator (when we collect) does need to map it's metric names (usually based on instrument info and/or view info). But, it does not HAVE an instrument.

Comment (cijothomas) [reply src/OpenTelemetry/Metrics/AggregatorStore.cs:110]: > I do not agree that an Instrument is part of the Aggregator

Nobody said otherwise.

**Comment by cijothomas** [comment]
Closing old PRs, as this is no longer relevant.

**Pull request closed without merging**